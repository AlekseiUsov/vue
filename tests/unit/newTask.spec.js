import { mount } from "@vue/test-utils";
import newTask from "../../src/components/atoms/ToDoNewTask.vue";

describe("ToDoNewTask.vue", () => {
  it("тестируем отрисовку компонента", async () => {
    const wrapper = mount(newTask);
    expect(wrapper).toMatchSnapshot();
  });

  it("тестируем отправку данных", async () => {
    const wrapper = mount(newTask);

    await wrapper.find("input").trigger("keyup.enter");
    expect(wrapper.emitted("getInputValue")).toBeTruthy();
  });

  it("тестируем очистку формы", async () => {
    const wrapper = mount(newTask);

    await wrapper.find("input").trigger("keyup.enter");
    expect(wrapper.text()).toBe('');
  });
});
