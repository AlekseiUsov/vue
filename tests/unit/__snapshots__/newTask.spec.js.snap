// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoNewTask.vue тестируем отрисовку компонента 1`] = `<input placeholder="Add a new task">`;
